pkg/rpcclient/nns/contract_test.go

```diff
@@ -195,3 +223,135 @@ func TestGetAllRecordsExpanded(t *testing.T) {
 		Data: "cool",
 	}, vals[0])
 }
+
+func TestRoots(t *testing.T) {
+	ta := new(testAct)
+	nns := NewReader(ta, util.Uint160{1, 2, 3})
+	ta.err = errors.New("")
+	uid, _, err := nns.Roots()
+	require.Error(t, err)
+
+	ta.err = nil
+	ta.res = &result.Invoke{
+		State: "HALT",
+		Stack: []stackitem.Item{
+			stackitem.NewInterop(result.Iterator{
+				ID: &uid,
+			}),
+		},
+	}
+	_, _, err = nns.Roots()
+	require.Error(t, err)
+
+	// Session-based iterator.
+	sid := uuid.New()
+	ta.res = &result.Invoke{
+		Session: sid,
+		State:   "HALT",
+		Stack: []stackitem.Item{
+			stackitem.NewInterop(result.Iterator{
+				ID: &uid,
+			}),
+		},
+	}
+	uid, _, err = nns.Roots()
+	require.NoError(t, err)
+	require.Equal(t, sid, uid)
+}
+
+func TestUpdate(t *testing.T) {
+	ta := &testAct{}
+	nns := New(ta, util.Uint160{1, 2, 3})
+
+	nef := []byte{0x01, 0x02, 0x03}
+	manifest := "manifest data"
+
+	ta.err = errors.New("test error")
+	_, _, err := nns.Update(nef, manifest)
+	require.Error(t, err)
+
+	// Test successful update
+	ta.err = nil
+	ta.txh = util.Uint256{0x04, 0x05, 0x06}
+	txh, vub, err := nns.Update(nef, manifest)
+	require.NoError(t, err)
+	require.Equal(t, ta.txh, txh)
+	require.Equal(t, ta.vub, vub)
+}
+
+func TestAddRoot(t *testing.T) {
+	ta := &testAct{}
+	nns := New(ta, util.Uint160{1, 2, 3})
+
+	root := "example.root"
+
+	ta.err = errors.New("test error")
+	_, _, err := nns.AddRoot(root)
+	require.Error(t, err)
+
+	// Test success case
+	ta.err = nil
+	ta.txh = util.Uint256{0x07, 0x08, 0x09}
+	txh, vub, err := nns.AddRoot(root)
+	require.NoError(t, err)
+	require.Equal(t, ta.txh, txh)
+	require.Equal(t, ta.vub, vub)
+}
+
+func TestSetPrice(t *testing.T) {
+	ta := &testAct{}
+	nns := New(ta, util.Uint160{1, 2, 3})
+
+	priceList := []*big.Int{big.NewInt(100), big.NewInt(200)}
+
+	ta.err = errors.New("test error")
+	_, _, err := nns.SetPrice(priceList)
+	require.Error(t, err)
+
+	// Test success case
+	ta.err = nil
+	ta.txh = util.Uint256{0x0A, 0x0B, 0x0C}
+	txh, vub, err := nns.SetPrice(priceList)
+	require.NoError(t, err)
+	require.Equal(t, ta.txh, txh)
+	require.Equal(t, ta.vub, vub)
+}
```

Comment:
OK, I see that test is here. But it doesn't call `NewParameterFromValue`. `SetPrice` will fail on real invocation, we need to adjust the method itself to provide `[]any` to `SendCall` instead of `[]*big.Int`.

Reply:
so SetPrice should be: SetPrice(priceList []any) ?

Reply:
It should be `SetPrice(priceList []int64)`, and inside you should convert `[]int64` to `[]any`, I guess.